packages/uniforms/src/useField.tsx

```diff
@@ -81,14 +85,17 @@ export function useField<
   let initialValue: Value | undefined;
   let value: Value | undefined = props.value ?? valueFromModel;
 
-  if (value === undefined) {
-    value = context.schema.getInitialValue(name, props);
-    initialValue = value;
-  } else if (props.value !== undefined && props.value !== valueFromModel) {
-    initialValue = props.value;
-  }
-
   if (options?.initialValue !== false) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    onChangeCalled = useRef(false);
```

Comment:
I think that it'd be better to do something like this:
```ts
const usesInitialValue = options?.initialValue !== false;
const onChangeCalled = usesInitialValue ? useRef(false) : { current: false };
```

Then this code as well as the `onChange` handler will be easier.